src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -3017,7 +3017,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
       res
     }
 
-    override def isVarargs: Boolean = definitions.isVarArgsList(paramss.flatten)
+    override def isVarargs: Boolean = definitions.isVarArgsList(lastSuch(paramss)(!_.isEmpty))
```

Comment:
What is this used for?

I find it hard to believe that we want to consider
```scala
def foo(x: Int*)()
```
as varargs, but
```scala
def foo(x: Int*)(y: Int)
```
as not. So could this be one of 
```scala
definitions.isVarArgsList(paramss.last)
// or
paramss.exists(definitions.isVarArgsList)
```
?

Reply:
IIRC, the code of `definitions.isVarArgsList` only checks the last element of the flattened list.  Thus, it makes sense to only look at the last non-empty list.

Reply:
Yikes, that looks like a bug, but the method is not used anywhere in the compiler, it's only part of the `scala.reflect` API :laughing:

Reply:
Yes, I just noticed that. The method we use is called `isVarargsMethod`, which is a simple flags check.

Maybe just make `isVarargs` an alias? Either way, I doubt this is that performance-critical.

Reply:
`isVarargsMethod` is not about methods that have scala-style varargs, it's for the compiler-generated overload that is emitted with the java `ACC_VARARGS` flag.

Maybe the best way forward is to just deprecate this method?